script/publish.ts

```diff
@@ -118,30 +118,25 @@ interface IUploadResult {
 }
 
 function upload(assetName: string, assetPath: string) {
-  const s3Info = {
-    accessKeyId: process.env.S3_KEY,
-    secretAccessKey: process.env.S3_SECRET,
-  }
-  const s3 = new AWS.S3(s3Info)
-
-  const bucket = process.env.S3_BUCKET || ''
-  const key = `releases/${packageInfo.getVersion()}-${sha}/${assetName.replace(
-    / /g,
-    ''
-  )}`
-  const url = `https://s3.amazonaws.com/${bucket}/${key}`
-
-  const uploadParams = {
-    Bucket: bucket,
-    ACL: 'public-read',
-    Key: key,
-    Body: Fs.createReadStream(assetPath),
+  const azureBlobService = getAzureBlobService()
+
+  if (azureBlobService === null) {
+    throw new Error(
+      'Unable to obtain an azure blob service. Deployment aborting...'
+    )
   }
 
+  const container = process.env.AZURE_BLOB_CONTAINER || ''
+  const cleanAssetName = assetName.replace(/ /g, '')
```

Comment:
Just curious, why not using `replaceAll` with a string instead of a regex?

```suggestion
  const cleanAssetName = assetName.replaceAll(' ', '')
```

Reply:
Cause that is what was there already in the S3 implementation. :D Do you have a reason to want to change it?

Reply:
Ohh sorry, didn't see it. I just tend to avoid regex when possible: slower and more error prone. It's just a habit 😆 

Happy to keep it if it gives you peace of mind minimizing changes 🤣